src/API/Data/UnitsRepository.cs

```diff
@@ -122,20 +121,19 @@ public class UnitsRepository : DataRepository
             }
             else
             {
-                var unitMembers = db.UnitMembers
-                    .Include(um => um.Unit)
-                    .Where(um => um.UnitId == unit.Id);
-                
-                //Remove UnitMembers for unit
-                // TODO: It would be nice to record these related items to the log table, in case we ever need to walk it back.
-                db.UnitMembers.RemoveRange(unitMembers);
+                //Remove, and log, UnitMembers and SupportRelationships for this unit.
+                var unitAndRelated = db.Units
+                    .Include(u => u.Parent)
+                    .Include(u => u.UnitMembers).ThenInclude(um => um.Person)
+                    .Include(u => u.SupportRelationships).ThenInclude(sr => sr.Department)
+                    .Single(u => u.Id == unit.Id);
 
-                var supportRelationships = db.SupportRelationships
-                    .Include(r => r.Unit)
-                    .Where(r => r.UnitId == unit.Id);
-                
+                //Write the logs with the whole enchilada 
+                LogPrevious(req, unitAndRelated);
+                //Remove UnitMembers for unit
+                db.UnitMembers.RemoveRange(unitAndRelated.UnitMembers);
                 //Remove SupportRelationships for unit
-                db.SupportRelationships.RemoveRange(supportRelationships);
+                db.SupportRelationships.RemoveRange(unitAndRelated.SupportRelationships);
```

Comment:
I've hit that problem with EF before, too, but I think we're avoiding it here.  It mostly comes up when you do something like:
```c#
var unit = db.Units.Include(u => u.SupportRelationships).Single(u => u.Id.Equals(1));
var supportRelationship = unit.SupportRelationships.Single(sr => sr.Id.Equals(23));
unit.SupportRelationships.Remove(supportRelationship);
db.SaveChanges();
```
I'll try to whip up a test to make sure we aren't creating orphaned relationships, though.  I think this may come up again when we start working on `unit_member_tools`.

Reply:
Added that test in 928447b.
EF, Postgres, and our database schema are all doing a good job protecting us from this problem.  The only relationship value I found that I could make null was  `unit_members.person_id`.  Everything else threw good errors when you tried to flip `unit_id` or `department_id` to null.

Reply:
That's a sweet relief. Thanks for doing this investigation!